test/utils/image/manifest_test.go

```diff
@@ -24,108 +24,86 @@ import (
 	"k8s.io/apimachinery/pkg/util/diff"
 )
 
-type result struct {
-	result string
-	err    error
-}
-
-var registryTests = []struct {
-	in  string
-	out result
-}{
-	{
-		"docker.io/library/test:123",
-		result{
-			result: "test.io/library/test:123",
-			err:    nil,
-		},
-	},
-	{
-		"docker.io/library/test",
-		result{
-			result: "test.io/library/test",
-			err:    nil,
-		},
-	},
-	{
-		"test",
-		result{
-			result: "test.io/library/test",
-			err:    nil,
-		},
-	},
-	{
-		"gcr.io/kubernetes-e2e-test-images/test:123",
-		result{
-			result: "test.io/kubernetes-e2e-test-images/test:123",
-			err:    nil,
-		},
-	},
-	{
-		"k8s.gcr.io/test:123",
-		result{
-			result: "test.io/test:123",
-			err:    nil,
-		},
-	},
-	{
-		"gcr.io/k8s-authenticated-test/test:123",
-		result{
-			result: "test.io/k8s-authenticated-test/test:123",
-			err:    nil,
-		},
-	},
-	{
-		"gcr.io/google-samples/test:latest",
-		result{
-			result: "test.io/google-samples/test:latest",
-			err:    nil,
-		},
-	},
-	{
-		"gcr.io/gke-release/test:latest",
-		result{
-			result: "test.io/gke-release/test:latest",
-			err:    nil,
-		},
-	},
-	{
-		"k8s.gcr.io/sig-storage/test:latest",
-		result{
-			result: "test.io/sig-storage/test:latest",
-			err:    nil,
-		},
-	},
-	{
-		"unknwon.io/google-samples/test:latest",
-		result{
-			result: "",
-			err:    fmt.Errorf("Registry: unknwon.io/google-samples is missing in test/utils/image/manifest.go, please add the registry, otherwise the test will fail on air-gapped clusters"),
-		},
-	},
-}
-
```

Comment:
Just shortened all this and put it in the test itself. The way it was written seems like those structures/values needed to be reused which isn't true and made this section harder to read.